src/uwtools/tests/config/test_support.py

```diff
@@ -51,3 +52,41 @@ def test_log_and_error(caplog):
         raise support.log_and_error(msg)
     assert msg in str(e.value)
     assert logged(caplog, msg)
+
+
+class Test_TaggedString:
+    """
+    Tests for class uwtools.config.support.TaggedString.
+    """
+
+    def comp(self, ts: support.TaggedString, s: str):
+        assert yaml.dump(ts, default_flow_style=True).strip() == s
+
+    @fixture
+    def loader(self):
+        yaml.add_representer(support.TaggedString, support.TaggedString.represent)
+        return YAMLConfig(config={})._yaml_loader
+
+    # These tests bypass YAML parsing, constructing nodes with explicit string values. They then
+    # demonstrate that those nodes' convert() methods return representations in type type specified
+    # by the tag.
+
+    def test_float_no(self, loader):
+        ts = support.TaggedString(loader, yaml.ScalarNode(tag="!float", value="foo"))
+        with raises(ValueError):
+            ts.convert()
+
+    def test_float_ok(self, loader):
+        ts = support.TaggedString(loader, yaml.ScalarNode(tag="!float", value="3.14"))
+        assert ts.convert() == 3.14
+        self.comp(ts, "!float '3.14'")
+
+    def test_int_no(self, loader):
+        ts = support.TaggedString(loader, yaml.ScalarNode(tag="!int", value="foo"))
+        with raises(ValueError):
+            ts.convert()
+
+    def test_int_ok(self, loader):
+        ts = support.TaggedString(loader, yaml.ScalarNode(tag="!int", value="88"))
+        assert ts.convert() == 88
+        self.comp(ts, "!int '88'")
```

Comment:
There's a weird behaving outlier that doesn't have a test, and I'm not sure what would happen. Python does not enjoy `int("3.14")`, but is fine with `int(3.14)`. I assume this might be a problem for a user at some point.

Reply:
I'm going to go ahead and merge this PR but will happily revisit this topic in the next PR if we think there's more to be done.